documentation/Linting-rules.md

```diff
@@ -0,0 +1,247 @@
+# CAMARA API Linting Rules Documentation
+
+## Introduction
+
+
+## 1. Spectral core ruleset
+
+Spectral has a built-in OpenAPI Specification ruleset that can be used to validate OpenAPI files.
+
+With `extends: "spectral:oas"` ("oas" being shorthand for OpenAPI Specification) in the ruleset file to rules for OpenAPI v2 and v3.x, depending on the appropriate OpenAPI version being used (this is automatically detected through formats) are added to the final ruleset.
+
+The full list of the rules in this ruleset are described in [OpenAPI Rules](https://docs.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules).
+
+
+`extends: [[spectral:oas, off]]`  - this avoids running any rules from the extended ruleset as they are disabled. Each rule then can be [enabled individually](https://docs.stoplight.io/docs/spectral/0a73453054745-recommended-or-all#enabling-rules).
+
+### Rule severity
+
+The `severity` keyword is optional in rule definition and can be `error`, `warn`, `info`, `hint`, or `off`.
+The default value is `warn`.

```

Comment:
We should discuss what is the expected outcome for each severity. For example should `warn` result in a change and re-lint, or can it be ignored?

Reply:
Good point. Maybe we can differentiate along the following lines:

* warnings can be ignored when introducing linting first time
  * OR (alternative) warnings can be ignored if correcting them would introduce a breaking change
* warnings shall not be introduced with a change (e.g. within a merge, that's the point where linting is done)

But that's to clarify in a next version. Will you open an issue for that @Kevsy ?